include/verilated_types.h

```diff
@@ -1322,6 +1322,29 @@ struct VlUnpacked final {
     WData* data() { return &m_storage[0]; }
     const WData* data() const { return &m_storage[0]; }
 
+    std::size_t size() const { return T_Depth; }
+    template <std::size_t CurrentDimension = 0>
+    int find_length(int dimension) const {
+        if (dimension == CurrentDimension) {
+            return size();
+        } else {
+            if constexpr (std::is_class_v<T_Value>) {
+                return m_storage[0].template find_length<CurrentDimension + 1>(dimension);
+            } else {
+                return size();
+            }
+        }
+    }
+    template <std::size_t CurrentDimension = 0>
+    auto& find_element(const std::vector<size_t>& indices) {
```

Comment:
This seems ok for multi-dimensional arrays, but once we have more complicated array-of-queue-with-array etc, is there a better way to structure it that is more "recursive-ish" that is, the caller strips dtype-by-dtype similar to how inside the Ast's there's a hierarchy of data types, rather than a single AstType that handles a multidimensional array?